Mds/new mobile sdk/okta 503880 revise quickstart (#3424)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/opensignin.md

Update the `signIn` function of `ContentView.swift` with the code to sign in the user:

```swift
func signIn() {
   Task {
      do {
         if let token = try await WebAuthentication.shared?.signIn(from: nil)
         {
            try Credential.store(token)
            updateStatus("Signed in", infoText: "", signedInStatus: true)
         }
      } catch {
         // Handle errors
         showError(title: "Unable to Sign In", error: error)
      }
   }
}
```

> The app hands the sign-in flow to the Okta SDK. The SDK presents the sign-in web view and waits for a result. If the sign-in is succesful, the app first stores the token using `Credential.store(_)`, and then updates the user interface by setting the `signedIn` state variable to `true`.

Reviewer: Is it worth mentioning here what the SDK displays if the sign-in is unsuccessful?

Reviewer: Or linking to relevant error handling content?

Author: For the first part, no as that's behavior that could change and has no impact on the code written by the developer.

I did think about looking for places to link to regarding error handling, but as a QuickStart I wanted to keep things to the minimum needed.

The other things is, link to what? If you do still think linking is needed, then what Okta content could I link to? In addition, on iOS the error code is either specific to Okta (defined in the SDK, in which case it's documented somewhere in the SDK so not linkable) or a generic iOS error such as a timeout, which is in the Apple documentation, (though specific error codes may be in different frameworks.) A more general "how to handle errors on iOS" also seems like something a developer should have access to on the Apple site... and external links are always a danger as things can change.

> The app hands the sign-in flow to the Okta SDK. The SDK presents the sign-in web view and waits for a result. If the sign-in is succesful, the app first stores the token using `Credential.store(_)`, and then updates the user interface by setting the `signedIn` state variable to `true`.

Reviewer 2: ```suggestion
The app hands the sign-in flow to the Okta SDK. The SDK presents the sign-in web view and waits for a result. If the sign-in flow is successful, the app first stores the token using `Credential.store(_)`, and then updates the user interface by setting the `signedIn` state variable to `true`.
```

Author: in the next commit

> The app hands the sign-in flow to the Okta SDK. The SDK presents the sign-in web view and waits for a result. If the sign-in is succesful, the app first stores the token using `Credential.store(_)`, and then updates the user interface by setting the `signedIn` state variable to `true`.

Reviewer 3: sign-in shouldn't be used as a noun acc2 the style guide.

```suggestion
The app hands the sign-in flow to the Okta SDK. The SDK presents the sign-in web view and waits for a result. If the sign-in attempt is successful, the app first stores the token using `Credential.store(_)`, and then updates the user interface by setting the `signedIn` state variable to `true`.
```

Author: Done. In the next commit.

> The app hands the sign-in flow to the Okta SDK. The SDK presents the sign-in web view and waits for a result. If the sign-in flow is successful, the app first stores the token using `Credential.store(_)`, and then updates the user interface by setting the `signedIn` state variable to `true`.

Reviewer 4: ```suggestion
The app hands the sign-in flow to the Okta SDK. The SDK presents the sign-in web view and waits for a result. If the sign-in flow is successful, the app first stores the token using `Credential.store(_:)`, and then updates the user interface by setting the `signedIn` state variable to `true`.
```

Author: Nice catch. It's in the next commit.

Author: I'm adding that wording to a commit, so please let me know if it's OK

> The stored token is used to check if the user is already signed in when the app is launched, and for API calls to the Okta org and authorization servers.

Reviewer 2: ```suggestion
The stored token is used to check if the user is already signed in when the app is launched and for API calls to the Okta org and authorization servers.
```

Author: Thanks. In the next commit.

> Remove the stored credential when the user signs-out. Replace the `signOut` function with the following code:

Author: Thanks. In the next commit.
